test/e2e/monitoring_stack_controller_test.go

```diff
@@ -511,12 +545,17 @@ func newAlerts(t *testing.T) *monv1.PrometheusRule {
 	return rule
 }
 
-func newMonitoringStack(t *testing.T, name string) *stack.MonitoringStack {
+func newMonitoringStack(t *testing.T, name string, disableAlertManager bool) *stack.MonitoringStack {
```

Comment:
(nit) to avoid grow the list of arguments indefinitely, I'd recommend using something like the functional options pattern.

```golang
func newMonitoringStack(t *testing.T, name string, options ...func(*stack.MonitoringStack)) *stack.MonitoringStack {
    ms := &stack.MonitoringStack{
    ...
    }
    for _, opt := range opts {
        opt(ms)
    }
    ...
    return ms
}
```

Then use it like

```
newMonitoringStack(t, name,
    func(ms *stack.MonitoringStack) {
        ms.Spec.AlertmanagerConfig.DisableAlertmanager = true
    },
)
```

Reply:
calling `newMonitoringStack(t, "foobar", false)` is confusing to read. [The infamous boolean trap!](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap)

while `newMontoringStack(t, "foobar", withoutAlertManager)` is clearer 
where `func withoutAlertManager(ms *stack.MonitoringStack) { ms.Spec.AlertmanagerConfig.Disabled = true }`